crates/bevy-inspector-egui/src/bevy_inspector/mod.rs

```diff
@@ -237,86 +237,80 @@ pub fn ui_for_state<T: FreelyMutableState + Reflect>(world: &mut World, ui: &mut
-            });
-            ui.checkbox(&mut is_fuzzy, "Fuzzy Match");
-            ui.memory_mut(|mem| {
-                *mem.data.get_persisted_mut_or_default(filter_kind_id) = is_fuzzy;
-            });
-            is_fuzzy
-        };
-
-        Filter { word, is_fuzzy }
-    }
-
-    /// empty filter which does nothing
-    fn empty() -> Self {
-        Self {
-            word: String::from(""),
-            is_fuzzy: false,
-        }
-    }
+/// Display all entities matching the static [`QueryFilter`]
+#[deprecated(since = "0.28.1", note = "use ui_for_entities_filtered instead")]
+pub fn ui_for_world_entities_filtered<QF: WorldQuery + QueryFilter>(
+    world: &mut World,
+    ui: &mut egui::Ui,
+    with_children: bool,
+) {
+    ui_for_entities_filtered::<QF>(world, ui, with_children);
+}
 
-    /// filter entities based on internal state
-    fn filter_entities(&self, world: &mut World, entities: &mut Vec<Entity>) {
-        if self.word.is_empty() {
-            return;
-        }
+/// Display all entities matching the static [`QueryFilter`]
+pub fn ui_for_entities_filtered<QF: WorldQuery + QueryFilter>(
+    world: &mut World,
+    ui: &mut egui::Ui,
+    with_children: bool,
+) {
+    ui_for_entities_filtered_with_filter::<QF, _>(world, ui, with_children, &Filter::empty());
+}
 
-        entities.retain(|entity| {
-            self_or_children_satisfy_filter(world, *entity, self.word.as_str(), self.is_fuzzy)
-        });
-    }
+/// Display all entities matching [`Without<Parent>`] and the provided [`EntityFilter`]
+pub fn ui_for_entities_with_filter<F: EntityFilter>(
+    world: &mut World,
+    ui: &mut egui::Ui,
+    with_children: bool,
+    filter: &F,
+) {
+    ui_for_entities_filtered_with_filter::<Without<Parent>, F>(world, ui, with_children, &filter);
 }
 
-/// Display all entities matching the given filter
-pub fn ui_for_world_entities_filtered<F: WorldQuery + QueryFilter>(
+/// Display all entities matching the static [`QueryFilter`]
+///
+/// Includes basic [`EntityFilter`]
+pub fn ui_for_entities_filtered_with_default_filter<QF: WorldQuery + QueryFilter>(
     world: &mut World,
     ui: &mut egui::Ui,
     with_children: bool,
 ) {
-    let filter = Filter::from_ui(ui);
+    let filter = Filter::from_ui(ui, egui::Id::new("default_world_entities_filter"));
+    ui_for_entities_filtered_with_filter::<QF, _>(world, ui, with_children, &filter);
+}
 
+/// Display all entities matching the given static [`QueryFilter`] and provided [`EntityFilter`]
+pub fn ui_for_entities_filtered_with_filter<QF, F>(
+    world: &mut World,
+    ui: &mut egui::Ui,
+    with_children: bool,
+    filter: &F,
+) where
+    QF: WorldQuery + QueryFilter,
+    F: EntityFilter,
+{
     let type_registry = world.resource::<AppTypeRegistry>().0.clone();
     let type_registry = type_registry.read();
 
-    let mut root_entities = world.query_filtered::<Entity, F>();
+    let mut root_entities = world.query_filtered::<Entity, QF>();
```

Comment:
This should probably follow the same method Hierarchy is doing: `<Entity, (Without<Parent>, QF)>` ?

Or should I remove  `Without<Parent>` from all "overloads" above?